backend/src/test/java/com/github/devbk84/backend/service/ProductServiceTest.java

```diff
@@ -53,4 +54,46 @@ void getProductsById(){
         assertEquals(expected, actual);
         verify(productRepo).findById("69");
     }
+
+    @Test
+    void deleteEntryByID(){
+        Product expectedProduct = new Product(
+                "8",
+                "Haferdrink",
+                "Test",
+                "orderFav",
+                new BigDecimal("1.50")
+        );
+
+        doNothing().when(productRepo).deleteById(isA(String.class));
+        productRepo.deleteById(expectedProduct.id());
+        verify(productRepo,times(1)).deleteById(expectedProduct.id());
+    }
+
+    @Test
+    void saveEntry() {
+        // GIVEN
+        Product expectedProduct = new Product(
+                "84",
+                "Milk",
+                "Test",
+                "Test",
+                new BigDecimal("1.50")
+        );
+
+        ProductDTO productToSave = new ProductDTO(
+                "Milk",
+                "Test",
+                "Test",
+                new BigDecimal("1.50")
+        );
+        // WHEN
+        when(productRepo.save(expectedProduct)).thenReturn(expectedProduct);
+        when(idService.generateID()).thenReturn("84");
```

Comment:
Wäre gut die ID in einem eigener Variable in der Methode zu speichern, damit du sie hier und in expectedProduct  zuweisen kannst (und sicher bist, das vom selben Wert gesprochen wird)

Man könnte in diesem Fall noch damit leben, wenn man's nicht macht, aber ist ne gute Angewohnheit!